### PR TITLE
[feat][gluon][mla] support MTP for mla

### PR DESCRIPTION
**Pull request by yanxuer-999**
## Motivation
Extend the gluon MLA decode kernel to support Multi-Token Prediction (MTP), i.e.
query length (qlen) > 1.

## Technical Details
- Add MTP (qlen 1–17) to `mla_decode_gluon` via a **grid-axis** path: `q_pos` is
  carried on a dedicated grid dimension, so each query position is its own program
  and reuses the existing per-regime layouts unchanged (no new MFMA/layout code).
  `IS_CAUSAL` gates a per-`q_pos` **causal tail mask** (row `q_pos` attends KV
  `[0, seq_len-qlen+q_pos]`); `qlen==1` keeps the original decode code path.
- One compiled kernel serves any runtime qlen (`q_pos` stride is 0 when `qlen==1`).
  4-D `[batch, qlen, nhead, dim]` inputs enable MTP; 

## Test Plan
- Correctness via `test_mla.py` (gluon vs torch golden, `checkAllclose`) over
  batch 1–256, nhead {4,8,12,16,64,128}, ctx up to 32768, qlen 1–17, bf16
  (+ fp8-KV and return-LSE).
- Perf: gluon latency (`run_perftest`) on the customer matrix
  (batch 1–4, nhead 16, ctx=16384, qlen 1/4/8/17, bf16).

## Test Result
Triton cache cleared before the run. gfx950 (MI350X), `test_mla.py`, all
`checkAllclose` PASS. Throughput speedup vs qlen=1 = `qlen × lat(q1) / lat(qlen)`.

### gluon MLA decode — nhead=16, ctx=16384, bf16
**batch = 1**
| qlen | latency (us) | throughput speedup vs qlen=1 |
|------|--------------|------------------------------|
| 1  | 50.17 | 1.00x  |
| 4  | 36.20 | 5.54x  |
| 8  | 33.27 | 12.06x |
| 17 | 47.37 | 18.01x |

**batch = 2**
| qlen | latency (us) | throughput speedup vs qlen=1 |
|------|--------------|------------------------------|
| 1  | 53.69 | 1.00x  |
| 4  | 33.97 | 6.32x  |
| 8  | 43.18 | 9.95x  |
| 17 | 77.63 | 11.76x |

**batch = 4**
| qlen | latency (us) | throughput speedup vs qlen=1 |
|------|--------------|------------------------------|
| 1  | 41.34 | 1.00x |
| 4  | 44.72 | 3.70x |
| 8  | 70.13 | 4.72x |
| 17 | 160.23 | 4.39x |

### Functional regression (all gluon paths PASS)
| Coverage | Config | Result |
|---|---|---|
| Small nhead MTP | nhead {4,8,12,16}, batch {1,4,16}, qlen {2,3,5,8}, ctx=16384, bf16 | PASS |
| Large ctx | nhead 16, batch {1,2}, qlen {1,4}, ctx=32768, bf16 | PASS |
| Large batch / large nhead (`bh64`) | nhead {64,128}, batch {64,128,256}, qlen=1, ctx=16384, bf16 | PASS |
| fp8 KV (`bh16bn128`) | nhead 16, batch 1, qlen 1, ctx=16384 | PASS |
| return-LSE | nhead 16, batch 4, qlen 4, ctx=16384, bf16 | PASS |

## Submission Checklist
- [ ] Look over the contributing guidelines at

### CODE REVIEW

File: aiter/ops/triton/gluon/mla_decode_gluon.py [hunk 1]
```diff
@@ -64,22 +64,28 @@ def _mla_decode_gluon(
     O,  # noqa: E741
     sm_scale,
     kv_scale,
+    QLEN,  # MTP query length (runtime); 1 for plain decode
```
Comment (Dewei-Wang-sh) [inline aiter/ops/triton/gluon/mla_decode_gluon.py]: constexpr and duplicate with is_causal

File: aiter/ops/triton/gluon/mla_decode_gluon.py [hunk 2]
```diff
@@ -144,9 +157,22 @@ def _mla_decode_gluon(
...
+    if IS_CAUSAL:
+        score_end = gl.minimum(split_kv_end, cur_batch_seq_len - QLEN + q_pos + 1)
+    else:
+        score_end = split_kv_end
+
     ######### layout setting begin #########
     # Q-side layouts + mfma_layout: switch by BLOCK_H.
-    # bh64 has BLOCK_H=64; bh16bn128 and bh16bn64 share BLOCK_H=16 (identical Q layouts + mfma orientation).
+    # bh64 has BLOCK_H=64; bh16bn128 and bh16bn64 share BLOCK_H=16
+    # (identical Q layouts + mfma orientation).
```
Comment (Dewei-Wang-sh) [inline aiter/ops/triton/gluon/mla_decode_gluon.py]: keep original

File: aiter/ops/triton/gluon/mla_decode_gluon.py (hunk 1 above)
Comment (yanxuer-999) [reply aiter/ops/triton/gluon/mla_decode_gluon.py]: the is_casual constexpr is removed， thanks for checking.

File: aiter/ops/triton/gluon/mla_decode_gluon.py (hunk 2 above)
Comment (yanxuer-999) [reply aiter/ops/triton/gluon/mla_decode_gluon.py]: done

File: aiter/ops/triton/gluon/mla_gluon.py [hunk 3]
```diff
@@ -984,22 +1082,28 @@ def mla_gluon(
...
         o,
         final_lse,
         seq_info,
-        logits_buf.stride(0),
-        logits_buf.stride(1),
-        logits_buf.stride(2),
+        sl_b,
+        sl_qs,
+        sl_h,
+        sl_split,
```
Comment (Dewei-Wang-sh) [inline aiter/ops/triton/gluon/mla_gluon.py:1097]: stride_logit_xx would be better, can change in next pr.

**Review by Dewei-Wang-sh — DISMISSED** [review]
lgtm

File: aiter/ops/triton/gluon/mla_gluon.py (hunk 3 above)
Comment (yanxuer-999) [reply aiter/ops/triton/gluon/mla_gluon.py:1097]: will fix this in nhead=96 pr

**Review by Dewei-Wang-sh — APPROVED** [review]
lgtm
